monai/losses/unified_focal_loss.py

```
@@ -39,119 +38,200 @@ def __init__(
+            y_pred = torch.cat([1 - y_pred, y_pred], dim=1)
+            is_already_prob = True
+            if y_true.shape[1] == 1:
+                y_true = one_hot(y_true, num_classes=2)
+        else:
+            is_already_prob = False
+
         n_pred_ch = y_pred.shape[1]
 
         if self.to_onehot_y:
             if n_pred_ch == 1:
                 warnings.warn("single channel prediction, `to_onehot_y=True` ignored.")
```

Ruff (0.14.10)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)